src/app/qgsmaptooldigitizefeature.cpp

```diff
@@ -171,11 +181,11 @@ void QgsMapToolDigitizeFeature::cadCanvasReleaseEvent( QgsMapMouseEvent *e )
       QgsGeometry g;
       if ( layerWKBType == QgsWkbTypes::Point )
       {
-        g = QgsGeometry::fromPointXY( savePoint );
+        g = QgsGeometry( qgis::make_unique<QgsPoint>( savePoint ) );
       }
       else if ( !QgsWkbTypes::isMultiType( layerWKBType ) && QgsWkbTypes::hasZ( layerWKBType ) )
       {
-        g = QgsGeometry( new QgsPoint( QgsWkbTypes::PointZ, savePoint.x(), savePoint.y(), defaultZValue() ) );
+        g = QgsGeometry( new QgsPoint( QgsWkbTypes::PointZ, savePoint.x(), savePoint.y(), isMatchPointZ ? savePoint.z() : defaultZValue() ) );
```

Comment:
```suggestion
        g = QgsGeometry( new QgsPoint( savePoint ) );
```

Reply:
OK to use make_unique, but no for the second suggestion since you have to deal with default Z value for this case.

Reply:
I see, let's keep it like this.

Note to self: move default value handling from "everywhere in the codebase" to "always use nan and do magic-value-mapping in providers".